src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -133,15 +133,23 @@ public function validate($entity, Constraint $constraint)
          */
         if ($result instanceof \Iterator) {
             $result->rewind();
-        } elseif (is_array($result)) {
+            if ($result instanceof \Countable && 1 < \count($result)) {
+                $result = array($result->current(), $result->current());
+            } else {
+                $result = $result->current();
+                $result = null === $result ? array() : array($result);
```

Comment:
Should we also account for iterators that have more than one result?

Reply:
I fear this could consume the iterator, could be an issue with non-rewindable ones

Reply:
hm indeed, we should rather not break that, might be a good idea to deprecate some of the edge cases we currently try to handle